packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Snippets/Snippet.DeclarationStatements.cs

```diff
@@ -52,6 +61,16 @@ public static MethodBodyStatement UsingDeclare<T>(string name, ScopedApi<T> valu
             return UsingDeclare(variableExpression, value);
         }
 
+        public static MethodBodyStatement Declare<T>(string name, T value, out T variable) where T : ScopedApi
+        {
+            var declaration = new CodeWriterDeclaration(name);
+            var variableExpression = new VariableExpression(TypeReferenceExpression.GetTypeFromDefinition(value.Type)!, declaration);
+
+            // TODO: need to constraint that subtype should have the corresponding constructor
+            variable = (T)Activator.CreateInstance(value.GetType(), variableExpression)!;
+            return new DeclarationExpression(variableExpression).Assign(value).Terminate();
+        }
```

Comment:
do we still need this magic?

Reply:
I think so, we need the concrete instance here.